### ISSUE TITLE
crash in TestEigh::test_value_subsets

### ISSUE THREAD

**Issue by rgommers** [scipy/scipy#11770]
On Arch Linux, GCC 9.3.0, built against NumPy 1.18.1 from Anaconda defaults; using MKL:

```
tests/test_decomp.py::TestEigh::test_wrong_inputs PASSED                         [ 13%]
tests/test_decomp.py::TestEigh::test_nonpositive_b PASSED                        [ 13%]
tests/test_decomp.py::TestEigh::test_value_subsets Fatal Python error: Aborted

Current thread 0x00007f9f3d00a740 (most recent call first):
  File "/home/rgommers/code/scipy/scipy/linalg/tests/test_decomp.py", line 826 in test_value_subsets
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/_pytest/python.py", line 176 in pytest_pyfunc_call
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/_pytest/python.py", line 1445 in runtest
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/_pytest/runner.py", line 126 in pytest_runtest_call
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/_pytest/runner.py", line 202 in <lambda>
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/_pytest/runner.py", line 229 in from_call
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/_pytest/runner.py", line 202 in call_runtest_hook
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/_pytest/runner.py", line 177 in call_and_report
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/_pytest/runner.py", line 96 in runtestprotocol
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/_pytest/runner.py", line 81 in pytest_runtest_protocol
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/_pytest/main.py", line 264 in pytest_runtestloop
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/_pytest/main.py", line 240 in _main
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/_pytest/main.py", line 196 in wrap_session
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/_pytest/main.py", line 233 in pytest_cmdline_main
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/rgommers/anaconda3/lib/python3.7/site-packages/_pytest/config/__init__.py", line 92 in main
  File "/home/rgommers/code/scipy/scipy/_lib/_testutils.py", line 67 in __call__
  File "<string>", line 1 in <module>
Aborted (core dumped)
```

Most likely due to recent `eigh` rewrite, although I haven't bisected to confirm.

**Comment by ilayn** [comment]
Grrr another one with MKL.... Related to #11709

**Comment by rgommers** [comment]
At least this one crashes consistently rather than 1 out of 10 times as in gh-11709

**Comment by ilayn** [comment]
The question is why others did not report this with MKL. But this gives me a bit of a hint though. Let's see.

**Comment by rgommers** [comment]
It only crashed for the `test('full')`, I don't think many people run that with master + MKL

**Comment by ilayn** [comment]
@congma , @rlucas7  Could you please also try to replicate this ?

**Comment by congma** [comment]
> @congma , @rlucas7 Could you please also try to replicate this ?

Yes. And this was also fixed after applying #11737 for me. What happens if you apply that one too?

Also, as I wrote in #11709, #11737 is a temporary stopgap. As @ilayn wrote there, it seems that the MKL `zheevr()` implementation is faulty in that it appears to take a wrong branch and overwrite the `isuppz` buffer when it doesn't need to.

**Comment by congma** [comment]
I tested with "full" mode again. Without #11737 (ie., with bare `master` currently at 7ac26a17), the test `scipy/linalg/tests/test_decomp.py::TestEigh::test_value_subsets` fails randomly with segfaults. With the patch, the test passes.

**Comment by rgommers** [comment]
Thanks @congma! I had missed that PR so far. I confirm that it fixes this issue for me, on Linux with both MKL `2019.4-243` and `2020.0-166`.

**Comment by ilayn** [comment]
Fixed in #11737
